aws/resource_aws_s3_bucket_object.go

```diff
@@ -385,6 +385,11 @@ func resourceAwsS3BucketObjectDelete(d *schema.ResourceData, meta interface{}) e
 
 	bucket := d.Get("bucket").(string)
 	key := d.Get("key").(string)
+	// We are effectively ignoring any leading '/' in the key name as aws.Config.DisableRestProtocolURICleaning is false
+	// so we need to explicitly ignore any leading '/' in the s3.ListObjectVersions call.
+	if strings.HasPrefix(key, "/") {
```

Comment:
FYI this will likely be updated as part of #7240 to just use `strings.TrimPrefix()` -- see also https://staticcheck.io/docs/checks#S1017

Reply:
👍